CodeGeneration/Sources/SyntaxSupport/Node.swift

```diff
@@ -380,3 +391,8 @@ fileprivate extension Child {
 fileprivate extension Node {
 
 }
+
```

Comment:
TODO: This needs a comment
TODO: Move `NodeInitRule` into it's own file.

Validation rules can live within NodeInitRule — perhaps in it's `init()` or in a separate function, i.e. `validate(for node: Node)`.